cfme/tests/v2v/test_v2v_migrations_ui.py

```diff
@@ -13,395 +15,322 @@
-    mapping = infrastructure_mapping_collection.create(form_data_single_datastore)
-    view = navigate_to(infrastructure_mapping_collection, 'All')
-    if appliance.version >= '5.10':  # As mapping could be on any page due to pagination
-        infrastructure_mapping_collection.find_mapping(mapping)  # We need to find mapping first
-    soft_assert(mapping.name in view.infra_mapping_list.read())
+    map_data = infra_mapping_default_data(source_provider, provider)
+    map_collection = appliance.collections.v2v_infra_mappings
+    mapping = map_collection.create(**map_data)
+
+    @request.addfinalizer
+    def _cleanup():
+        map_collection.delete(mapping)
+    view = navigate_to(map_collection, "All")
     mapping_list = view.infra_mapping_list
+
+    # Test1: Check custom map in infra map list
+    soft_assert(mapping.name in mapping_list.read())
+
+    # Test2: Validate infra map description
     soft_assert(str(mapping_list.get_map_description(mapping.name)) == mapping.description)
-    soft_assert(mapping.form_data['cluster']['mappings'][0]['sources'][0].format() in
-        mapping_list.get_map_source_clusters(mapping.name)[0])
-    soft_assert(mapping.form_data['cluster']['mappings'][0]['target'][0].format() in
-     mapping_list.get_map_target_clusters(mapping.name)[0])
-    soft_assert(mapping.form_data['datastore'].values()[0]['mappings'][0]['sources'][0].format() in
-     mapping_list.get_map_source_datastores(mapping.name)[0])
-    soft_assert(mapping.form_data['datastore'].values()[0]['mappings'][0]['target'][0].format() in
-     mapping_list.get_map_target_datastores(mapping.name)[0])
-    soft_assert(mapping.form_data['network'].values()[0]['mappings'][0]['sources'][0].format() in
-     mapping_list.get_map_source_networks(mapping.name)[0])
-    soft_assert(mapping.form_data['network'].values()[0]['mappings'][0]['target'][0].format() in
-     mapping_list.get_map_target_networks(mapping.name)[0])
-
-    # Testing if refreshing hosts cause any "Network Missing"
-    rhv_prov = v2v_providers.rhv_provider
-    host = rhv_prov.hosts.all()[0]
-    host.refresh(cancel=True)
-    view = navigate_to(infrastructure_mapping_collection, 'All')
-    mapping_list = view.infra_mapping_list
-    soft_assert(mapping_list.get_map_source_networks(mapping.name) != [])
-    soft_assert(mapping_list.get_map_target_networks(mapping.name) != [])
 
-    mapping_list.delete_mapping(mapping.name)
-    view.browser.refresh()
-    view.wait_displayed()
-    try:
-        assert mapping.name not in view.infra_mapping_list.read()
-    except NoSuchElementException:
-        # meaning there was only one mapping that is deleted, list is empty
-        pass
+    # Test3: Source cluster from UI
+    soft_assert(mapping.clusters[0].sources[0].format() in
+                mapping_list.get_map_source_clusters(mapping.name)[0])
 
+    # Test4: Target cluster from UI
+    soft_assert(mapping.clusters[0].targets[0].format() in
```

Comment:
What's the purpose of these empty `.format()` calls?

Reply:
if you see value of `mapping.clusters[0].targets[0]` it is in `partial_match` format.. so to convert it into normal data we need this `.format()`